index/scorch/snapshot_index.go

```diff
@@ -37,6 +38,7 @@ import (
 // re usable, threadsafe levenshtein builders
 var lb1, lb2 *lev.LevenshteinAutomatonBuilder
 
+type diskStatsReporter segment.DiskStatsReporter
```

Comment:
Is this step necessary, can't we just type assert to segment.DiskStatsReporter?

Reply:
Yeah at a couple of places, there are library name and variable name conflicts, for eg. `indexSnapshot.segment.segment `.

Reply:
Ah variable and import conflict. Should update the variable name then (it looks quite confusing :)). I'll make that change separately.